README.md

```diff
@@ -340,6 +327,61 @@ Note how this example responds to various types of navigations:
 
 Notice also how by passing through the `AbortSignal` found in `e.signal`, we ensure that any aborted navigations abort the associated fetch as well.
 
+#### Example: async transitions with special back/forward handling
+
+Sometimes it's desirable to handle back/forward navigations specially, e.g. reusing cached views by transitioning them onto the screen. This can be done by branching as follows:
+
+```js
+appHistory.addEventListener("navigate", e => {
+  // As before.
+  if (!e.canRespond || e.hashChange) {
+    return;
+  }
+
+  if (myFramework.currentPage) {
+    await myFramework.currentPage.transitionOut();
+  }
+
+  const isBackForward = appHistory.entries.includes(event.destination);
+  let { key } = event.destination;
+
+  if (isBackForward && myFramework.previousPages.has(key)) {
+    await myFramework.previousPages.get(key).transitionIn();
+  } else {
+    // This will probably result in myFramework storing the rendered page in myFramework.previousPages.
+    await myFramework.renderPage(event.destination);
+  }
+});
+```
+
+#### Restrictions on firing, canceling, and responding
+
+There are many types of navigations a given page can experience; see [this appendix](#appendix-types-of-navigations) for a full breakdown. Some of these need to be treated specially for the purposes of the navigate event.
+
+First, the following navigations **will not fire `navigate`** at all:
+
+- User-initiated [cross-document](#appendix-types-of-navigations) navigations via browser UI, such as the URL bar, back/forward button, or bookmarks.
```

Comment:
I may be reading this wrong, but you list `user-initiated cross-document navigations via... back/forward button` as something that will never fire `navigate` here. 

yet below on line 372 it says `user-initiated navigations (cross- or same-document) via browser's back/foward buttons` which implies that navigate is fired (else there would be no need to mention `event.preventDefault()` ?)

also the table appears to list browser back/forward as firing navigate.

Reply:
Good catch. Listing cross-document browser back/forward navigations as non-cancelable is redundant since they won't fire navigate at all. Fixed.

As for the table, it says "Yes †*" where - † = No if cross-document (and \* = No if the URL differs in components besides path/fragment/query). I struggled with these "Yes" table entries; most of them are "Yes except in specific rarer circumstances", but that cell has more serious exceptions. Any suggestions are welcome.